core/trino-spi/src/main/java/io/trino/spi/predicate/Ranges.java

```diff
@@ -28,4 +28,9 @@
      * @return Single range encompassing all of allowed the ranges
      */
     Range getSpan();
+
+    /**
+     * Try to return a more compact representation (if possible).
+     */
+    ValueSet getCompactedRanges();
```

Comment:
Document whether the compacted version is supported to 

- be equal (as in `Object.equals`)
- contain same information, or can eg be slightly lossy (widening) for more efficient compaction
- what does is returned when compaction is not possible?

Reply:
Also, higher level

- why is this a method on `Ranges` instead of on `ValueSet`?
- why do we want to have such method at all? I would hope the `ValueSet` instance is "as compact as possible", always

Reply:
> Document whether the compacted version is supported to

Sounds good - done.

Reply:
Thinking about it, the relevant part of this PR (https://github.com/trinodb/trino/pull/3316/commits/23d6a6e91be03de75d0c5ed152447439e4de991b#diff-a3ede16a05943d869ba6d7b8821761bc3ed3d23915bb0b3bf8bac363b7038dea) is small enough to just move into our connector (after https://github.com/trinodb/trino/pull/9868 is merged).

Reply:
> I agree, but such compactions [may cause CBO to change the plan](https://github.com/trinodb/trino/pull/3316#issuecomment-608980527) (since we estimate the statistics differently depending on the predicate being a range or a discrete set of values)

Is this something that could be improved on?

cc @sopel39

Reply:
> Is this something that could be improved on?

IMO it should be, I don't think there should be code duality (DF vs CBO)